airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java

```diff
@@ -1600,6 +1602,74 @@ public boolean getConnectionHasAlphaOrBetaConnector(final UUID connectionId) thr
     return countResult > 0;
   }
 
+  public Optional<ConnectorBuilderProject> getConnectorBuilderProject(final UUID builderProjectId, final UUID workspaceId) throws IOException {
+    return database.query(ctx -> ctx.select(CONNECTOR_BUILDER_PROJECT.asterisk())
+        .from(CONNECTOR_BUILDER_PROJECT)
+        .where(getBuilderProjectIdCondition(builderProjectId, workspaceId))
+        .fetch())
+        .map(DbConverter::buildConnectorBuilderProject)
+        .stream()
+        .findFirst();
```

Comment:
If we only care about the first one (and hopefully only one), would this be better in terms of DB performance to have a `limit(1)` somewhere?

Reply:
There can't be more than one because the project id is the PK